build/lib/compilation.js

```diff
@@ -61,7 +61,9 @@ function createCompile(build, emitError) {
 function compileTask(out, build) {
     return function () {
         var compile = createCompile(build, true);
-        var src = es.merge(gulp.src('src/**', { base: 'src' }), gulp.src('node_modules/typescript/lib/lib.d.ts'));
+        var src = es.merge(gulp.src('src/**', { base: 'src' }), 
+        // gulp.src('node_modules/typescript/lib/lib.dom.d.ts'),
+        gulp.src('node_modules/typescript/lib/lib.es6.d.ts'));
```

Comment:
Careful with that because I believe it leaks globals into the CU that we cannot use, e.g `Symbol.iterator` etc. We need to configure the compiler so that it emits ES6 but doesn't use ES6 globals...

Reply:
Why can't we use that, is it not in electron's Node?

Reply:
The monaco editor supports IE11

Reply:
We must use a different compilation pass for the monaco editor.

Reply:
We have `src/tsconfig.monaco.json` which is *only* the monaco editor. Problem is that tsserver can only use one project file so that you won't see compile errors in VS Code, e.g. when using `Symbol.iterator` in `vs/base` or `vs/editor`. Running `tsc -p tsconifg.monaco.json` takes around 15secs which I'd say is too much for a pre-commit hook. It does already run travis et al but with little respect...

Reply:
**bold**